Translation of "Building on Lisk" into Indonesian (#264)
LiskHQ/lisk-docs: i18n/ind/docusaurus-plugin-content-docs/current/building-on-lisk/deploying-smart-contract/with-Hardhat.md

---
title: ... menggunakan Hardhat
slug: /building-on-lisk/deploying-smart-contract/with-Hardhat

> description: "Panduan untuk mendeploy smart contract di jaringan test Lisk menggunakan Hardhat. Termasuk instruksi untuk mengatur environment, mengompilasi, dan mendeploy smart contract."

Reviewer: ```suggestion
description: "Panduan untuk meluncurkan kontrak pintar di jaringan test Lisk menggunakan Hardhat. Termasuk instruksi untuk mengatur environment, mengompilasi, dan meluncurkan kontrak pintar."
```

keywords:
  [
    "Hardhat",

>     "smart contract",

Reviewer: ```suggestion
    "kontrak pintar",
```

    "ERC-721",
    "Lisk",

>     "Lisk test network",

Reviewer: ```suggestion
    "jaringan test Lisk",
```

>     "Lisk testnet",

Reviewer: ```suggestion
    "testnet Lisk",
```

>     "deploying smart contracts",

Reviewer: ```suggestion
    "meluncurkan kontrak pintar",
```

>     "build on lisk",

Reviewer: ```suggestion
    "membangun di lisk",
```

>     "write smart contract",

Reviewer: ```suggestion
    "menulis ke kontrak pintar",
```

>     "smart contract development",

Reviewer: ```suggestion
    "pengembangan kontrak pintar",
```

  ]
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

> # Mendeploy Smart Contract dengan Hardhat

Reviewer: ```suggestion
# Meluncurkan Kontrak Pintar dengan Hardhat
```

## Prasyarat

### Node v18+

Panduan ini memerlukan Node versi 18+ yang sudah terinstal.

- _Download_ [Node v18+](https://nodejs.org/en/download/)

Jika Anda menggunakan `nvm` untuk mengelola versi Node Anda, cukup jalankan perintah: `nvm install 18`.

> ### Dana _Wallet_

Reviewer: ```suggestion
### Dana Dompet
```

> **Mendeploy kontrak** ke blockchain memerlukan **biaya gas**.  

Reviewer: ```suggestion
**Meluncurkan kontrak** ke blockchain memerlukan **biaya gas**.  
```

> Oleh karena itu, Anda perlu mendanai _wallet_ Anda dengan ETH untuk menutupi biaya gas tersebut.

Reviewer: ```suggestion
Oleh karena itu, Anda perlu mendanai dompet Anda dengan ETH untuk menutupi biaya gas tersebut.
```

> Dalam panduan ini, Anda akan mendeploy kontrak ke Lisk Sepolia Testnet.

Reviewer: ```suggestion
Dalam panduan ini, Anda akan meluncurkan kontrak ke Lisk Sepolia Testnet.
```

Anda dapat men-_deposit_ token yang diperlukan menggunakan [Lisk Bridge](https://sepolia-bridge.lisk.com/bridge/lisk-sepolia-testnet).

> Jika _wallet_ Anda tidak memiliki cukup `SepoliaETH`, gunakan salah satu faucet yang tersedia untuk **Ethereum Sepolia** Testnet, seperti [https://sepoliafaucet.com](https://sepoliafaucet.com/) untuk menerima ETH Testnet secara gratis.  

Reviewer: ```suggestion
Jika dompet Anda tidak memiliki `SepoliaETH` yang cukup, gunakan salah satu faucet yang tersedia untuk **Ethereum Sepolia** Testnet, seperti [https://sepoliafaucet.com](https://sepoliafaucet.com/) untuk menerima ETH Testnet secara gratis.  
```

Kemudian, gunakan Lisk Bridge yang disebutkan sebelumnya untuk mengirim token dari **Ethereum Sepolia Testnet** ke **Lisk Sepolia Testnet**.

:::note

> Anda dapat mendeploy kontrak di Lisk Mainnet dengan menggunakan proses yang sama.  

Reviewer: ```suggestion
Anda dapat meluncurkan kontrak di Lisk Mainnet dengan menggunakan proses yang sama.  
```

> Untuk mendeploy ke mainnet, pastikan _wallet_ Anda memiliki cukup ETH.

Reviewer: ```suggestion
Untuk meluncurkan ke mainnet, pastikan dompet Anda memiliki cukup ETH.
```

Teks berikut mencakup perintah untuk Lisk dan Lisk Sepolia demi kemudahan Anda.  

> Untuk informasi lebih lanjut, lihat [jaringan Lisk yang tersedia](/network-info) dan [cara menghubungkan wallet ke jaringan tersebut](/user/connecting-to-a-wallet).

Reviewer: ```suggestion
Untuk informasi lebih lanjut, lihat [jaringan Lisk yang tersedia](/network-info) dan [cara menghubungkan dompet ke jaringan tersebut](/user/connecting-to-a-wallet).
```

:::

## Membuat Proyek

> Sebelum Anda dapat mulai mendeploy _smart contract_ ke Lisk, Anda perlu menyiapkan _environment_ pengembangan dengan membuat proyek Node.js.

Reviewer: ```suggestion
Sebelum Anda dapat mulai meluncurkan kontrak pintar ke Lisk, Anda perlu menyiapkan _environment_ pengembangan dengan membuat proyek Node.js.
```

✔ Do you want to install this sample project's dependencies with npm (@nomicfoundation/hardhat-toolbox)? (Y/n) · y
```

Proses penyiapan proyek akan memakan waktu beberapa saat hingga selesai.

## Mengonfigurasi Hardhat dengan Lisk

> Untuk mendeploy _smart contract_ ke jaringan Lisk, Anda perlu mengonfigurasi proyek Hardhat Anda dan menambahkan jaringan Lisk.

Reviewer: ```suggestion
Untuk meluncurkan kontrak pintar ke jaringan Lisk, Anda perlu mengonfigurasi proyek Hardhat Anda dan menambahkan jaringan Lisk.
```

Contoh ini menggunakan [dotenv](https://www.npmjs.com/package/dotenv) untuk memuat variabel environment `WALLET_KEY` dari file `.env` ke `process.env.WALLET_KEY`.  
Anda sebaiknya menggunakan metode serupa untuk menghindari menuliskan _private key_ secara langsung di dalam _source code_ Anda.

```bash
npm install --save-dev dotenv
```

Setelah Anda menginstal `dotenv`, buat file `.env` dengan konten berikut:

```
WALLET_KEY=<YOUR_PRIVATE_KEY>
```

> Ganti `<YOUR_PRIVATE_KEY>` dengan _private key_ _wallet_ Anda.

Reviewer: ```suggestion
Ganti `<YOUR_PRIVATE_KEY>` dengan _private key_ dompet Anda.
```

:::caution

> `WALLET_KEY` adalah _private key_ dari _wallet_ yang akan digunakan saat mendeploy kontrak.  

Reviewer: ```suggestion
`WALLET_KEY` adalah _private key_ dari dompet yang akan digunakan saat meluncurkan kontrak.  
```

> Ikuti instruksi dari _wallet_ Anda untuk mendapatkan _private key_.  

Reviewer: ```suggestion
Ikuti instruksi dari dompet Anda untuk mendapatkan _private key_.  
```

> Misalnya, untuk **MetaMask**, silakan ikuti [instruksi ini](https://support.metamask.io/hc/en-us/articles/360015289632-How-to-export-an-account-s-private-key).  

Reviewer 2: ```suggestion
Misalnya, untuk **MetaMask**, silakan ikuti [instruksi ini](https://support.metamask.io/configure/accounts/how-to-export-an-accounts-private-key/).  
```

    export default config;
    ```

  </TabItem>
</Tabs>

## Membuat Kontrak

> Untuk kemudahan dan keamanan, kita akan menggunakan _interface_ `ERC721` yang disediakan oleh [OpenZeppelin Contracts library](https://docs.openzeppelin.com/contracts/5.x/) untuk membuat _smart contract_ NFT.  

Reviewer: ```suggestion
Untuk kemudahan dan keamanan, kita akan menggunakan _interface_ `ERC721` yang disediakan oleh [OpenZeppelin Contracts library](https://docs.openzeppelin.com/contracts/5.x/) untuk membuat kontrak pintar NFT.  
```

Dengan OpenZeppelin, kita tidak perlu menulis keseluruhan _interface_ ERC-721. Sebaliknya, kita dapat mengimpor kontrak dari library dan menggunakan fungsinya.

> Untuk menambahkan OpenZeppelin Contracts library ke proyek Anda, jalankan perintah berikut:

Reviewer: ```suggestion
Untuk menambahkan library OpenZeppelin Contract ke proyek Anda, jalankan perintah berikut:
```

}
```

> ## Mengompilasi Smart Contract

Reviewer 2: There seems to be an inconsistency here; either use the term 'Contract' or 'Kontrak'. I would use Contract since that is widely known.

}
```

> ## Meng-_compile_ Smart Contract

Reviewer: ```suggestion
## Mengompilasi Kontrak Pintar
```

Untuk mengompilasi kontrak menggunakan Hardhat, cukup jalankan perintah berikut:

```bash
npx hardhat compile
```

Setelah kompilasi berhasil, Anda akan melihat folder baru bernama `artifacts/`, yang berisi [artefak kompilasi](https://hardhat.org/hardhat-runner/docs/advanced/artifacts).

> ## Mendeploy Smart Contract

Reviewer: ```suggestion
## Meluncurkan Kontrak Pintar
```

> Setelah kontrak Anda berhasil dikompilasi, Anda dapat mendeploy kontrak ke jaringan test Lisk Sepolia.

Reviewer: ```suggestion
Setelah kontrak Anda berhasil dikompilasi, Anda dapat meluncurkan kontrak ke jaringan test Lisk Sepolia.
```

> Untuk mendeploy kontrak ke jaringan test Lisk Sepolia, Anda perlu memodifikasi file `scripts/deploy.ts` di proyek Anda:

Reviewer: ```suggestion
Untuk meluncurkan kontrak ke jaringan test Lisk Sepolia, Anda perlu memodifikasi file `scripts/deploy.ts` di proyek Anda:
```

```ts title="scripts/deploy.ts"

> import { ethers } from "hardhat";

Reviewer: ```suggestion
import { ethers } from 'hardhat';
```

Reviewer 2: It should be single quotes, as it is in the English version.

async function main() {

>   const nft = await ethers.deployContract("NFT");

Reviewer: ```suggestion
  const nft = await ethers.deployContract('NFT');
```

Reviewer 2: It should be single quotes, as it is in the English version.

  await nft.waitForDeployment();

>   console.log("NFT Contract Deployed at " + nft.target);

Reviewer: ```suggestion
  console.log('NFT Contract Deployed at ' + nft.target);
```

Reviewer 2: It should be single quotes, as it is in the English version.

}

// Kami merekomendasikan pattern ini agar dapat menggunakan async/await di seluruh bagian kode.
// dan menangani error dengan baik.
main().catch((error) => {
  console.error(error);
  process.exitCode = 1;
});
```

> Anda juga memerlukan ETH Testnet di _wallet_ Anda.  

Reviewer: ```suggestion
Anda juga memerlukan ETH Testnet di dompet Anda.  
```

> Lihat bagian [Prasyarat](#prasyarat) jika Anda belum melakukannya. Jika tidak, upaya _deployment_ akan gagal.

Reviewer: ```suggestion
Lihat bagian [Prasyarat](#prasyarat) jika Anda belum melakukannya. Jika tidak, upaya peluncuran akan gagal.
```

  </TabItem>
</Tabs>

<!-- TODO: Add link to the block explorer section -->

> Kontrak akan dideploy di Lisk Sepolia Testnet.  

Reviewer: ```suggestion
Kontrak akan diluncurkan di Lisk Sepolia Testnet.  
```

> Anda dapat melihat status deployment dan kontrak dengan menggunakan _block explorer_ dan mencari alamat yang dikembalikan oleh skrip deploy Anda.

Reviewer: ```suggestion
Anda dapat melihat status peluncuran dan kontrak dengan menggunakan penjelajah blockchain dan mencari alamat yang dikembalikan oleh skrip peluncuran Anda.
```

> Jika Anda mendeploy kontrak baru atau yang dimodifikasi, Anda perlu memverifikasinya terlebih dahulu.

Reviewer: ```suggestion
Jika Anda meluncurkan kontrak baru atau yang dimodifikasi, Anda perlu memverifikasinya terlebih dahulu.
```

> ## Memverifikasi Smart Contract

Reviewer: ```suggestion
## Memverifikasi Kontrak Pintar
```

> Jika Anda ingin berinteraksi dengan kontrak Anda di _block explorer_, Anda atau orang lain perlu memverifikasinya terlebih dahulu.  

Reviewer: ```suggestion
Jika Anda ingin berinteraksi dengan kontrak Anda di penjelajah blockchain, Anda atau orang lain perlu memverifikasinya terlebih dahulu.  
```

> Kontrak di atas sudah diverifikasi, sehingga Anda seharusnya dapat melihat versi Anda di _block explorer_.  

Reviewer: ```suggestion
Kontrak di atas sudah diverifikasi, sehingga Anda seharusnya dapat melihat versi Anda di penjelajah blockchain.  
```

  </TabItem>
  <TabItem value="testnet" label="Lisk Sepolia" default>
  ```ts title="hardhat.config.ts"

>   // Tambahkan informasi berikut setelah konfigurasi "networks" di HardhatUserConfig:

Reviewer: ```suggestion
  // Tambahkan informasi berikut setelah konfigurasi "networks" di HardhatUserConfig
```

  </TabItem>
</Tabs>

> Anda akan melihat output yang serupa dengan berikut:

Reviewer: ```suggestion
Anda akan melihat output yang serupa dengan:
```

:::info

Anda tidak dapat memverifikasi ulang kontrak yang identik dengan yang sudah diverifikasi.  

> Jika Anda mencoba melakukannya, seperti memverifikasi kontrak di atas, Anda akan mendapatkan pesan yang seperti:

Reviewer: ```suggestion
Jika Anda mencoba melakukannya, seperti memverifikasi kontrak di atas, Anda akan mendapatkan pesan yang serupa dengan:
```

```text
The contract 0xC10710ac55C98f9AACdc9cD0A506411FBe0af71D has already been verified on Etherscan.
https://sepolia-blockscout.lisk.com/address/0xC10710ac55C98f9AACdc9cD0A506411FBe0af71D#code
```

:::

> Lihat kontrak Anda di BlockScout dengan mengikuti [link ke kontrak yang telah dideploy](https://sepolia-blockscout.lisk.com/address/0xC10710ac55C98f9AACdc9cD0A506411FBe0af71D?tab=contract) yang ditampilkan dalam pesan output langkah sebelumnya.  

Reviewer: ```suggestion
Lihat kontrak Anda di BlockScout dengan mengikuti [link ke kontrak yang telah diluncurkan](https://sepolia-blockscout.lisk.com/address/0xC10710ac55C98f9AACdc9cD0A506411FBe0af71D?tab=contract) yang ditampilkan dalam pesan output langkah sebelumnya.  
```

> Block explorer akan mengonfirmasi bahwa kontrak tersebut telah diverifikasi dan memungkinkan Anda untuk [berinteraksi](#berinteraksi-dengan-smart-contract) dengannya.

Reviewer: ```suggestion
Penjelajah Blockchain akan mengonfirmasi bahwa kontrak tersebut telah diverifikasi dan memungkinkan Anda untuk [berinteraksi](#berinteraksi-dengan-smart-contract) dengannya.
```

> ## Berinteraksi dengan Smart Contract

Reviewer: ```suggestion
## Berinteraksi dengan Kontrak Pintar
```

> Setelah [kontrak diverifikasi](#verifying-the-smart-contract), Anda dapat menggunakan tab `Read Contract` dan `Write Contract` untuk berinteraksi dengan kontrak yang telah dideploy melalui [BlockScout](https://sepolia-blockscout.lisk.com/address/0xC10710ac55C98f9AACdc9cD0A506411FBe0af71D?tab=contract).  

Reviewer: ```suggestion
Setelah [kontrak diverifikasi](#verifying-the-smart-contract), Anda dapat menggunakan tab `Read Contract` dan `Write Contract` untuk berinteraksi dengan kontrak yang telah diluncurkan melalui [BlockScout](https://sepolia-blockscout.lisk.com/address/0xC10710ac55C98f9AACdc9cD0A506411FBe0af71D?tab=contract).  
```

Jangan lupa untuk memperbarui alamat kontrak di URL BlockScout.  

> Anda juga perlu menghubungkan _wallet_ Anda terlebih dahulu dengan mengklik tombol `Connect Wallet`.

Reviewer: ```suggestion
Anda juga perlu menghubungkan dompet Anda terlebih dahulu dengan mengklik tombol `Connect Wallet`.
```
